### ISSUE TITLE
Unable to use JWKS with Keycloak 18

### ISSUE THREAD

**Issue by carlosli888** [keycloak/keycloak#12344]
### Describe the bug

I am unable to set up keys for clients to use with the client credentials flow with private_key_jwt method via the Keycloak administrator portal.

### Version

18

### Expected behavior

I am able to configure Keycloak to use the key I specified using the JWKS format.

### Actual behavior

[key-test-client.rs256.jwks.txt](https://github.com/keycloak/keycloak/files/8836127/key-test-client.rs256.jwks.txt)

**JWKS Import**
When I try to import the attached RS256 key via the Key tab, the import fails.
![image](https://user-images.githubusercontent.com/47931389/171964304-13ae8398-a20d-4e33-9286-a41fd585e711.png)

**Direct JWKS**
If I enter the contents of the attached RS256 JWKS into the JWKS setting, it is unable to find the key.  Actually, the console has no kids in storage.
![image](https://user-images.githubusercontent.com/47931389/171964441-b4bc8b9b-3c1d-4bd5-9212-a1282fa5a600.png)
![image](https://user-images.githubusercontent.com/47931389/171964544-7fc79596-0192-45b1-93b6-35de8327f858.png)

**JWKS URL**
If I put the attached RS256 key in a web server, Keycloak is also not able to find the key.  Actually, the console again shows that it has no kids in storage.
![image](https://user-images.githubusercontent.com/47931389/171964832-f995225d-779f-488c-a7ec-1c00367f5bba.png)
![image](https://user-images.githubusercontent.com/47931389/171964960-b0e01db2-8ec6-4a5e-8a60-3f58661aec6b.png)

### How to Reproduce?

1. Create a RSA private/public key pair.
```
openssl genrsa -out key-test-client.rs256.privatekey.pem
openssl rsa -in key-test-client.rs256.privatekey.pem -outform pem -pubout -out .\key-test-client.rs256.publickey.pem
```
2. Use website (https://8gwifi.org/jwkconvertfunctions.jsp) to convert the **public key** (in PEM format) to JWK.  DO NOT send your private key.  Keycloak only needs the public key to validate the JWT your client application creates and signs.  This is why only the public key is sent to the external party to convert it from the PEM format to JWK.
3. Create a JSON file that follows the JWKS format and add the resulting JWK above as a key in the "keys" array.
4. Save the file.
[key-test-client.rs256.jwks.txt](https://github.com/keycloak/keycloak/files/8836139/key-test-client.rs256.jwks.txt)
5. Log back in to the Keycloak admin tool as the administrator.
6. Select to edit the "key-test-client" client.
7. In the Credentials tab, ensure that the client is set to use a "Signed Jwt" and "RS256".
![image](https://user-images.githubusercontent.com/47931389/171965326-b5120094-3c6d-4a3d-9de5-d3c906c3dc4e.png)
8. In the Keys tab, select to use JWKS.
9. Paste the content of the file created earlier.
10. Save.
11. Try authenticating a client using client credentials grant and a jwt signed with the matching private key.
12. After authentication failed, check the Keycloak console for errors *(see actual behaviour section for screenshot)*.
13. Go back to the Keys tab for the client, and change it to use a JWKS URL.
14. Save the changes.
15. Repeat authentication test.
16. After authentication failed, check the Keycloak console for errors *(see actual behaviour section for screenshot)*.
17. Go back to the Keys tab for the client, and change it to not use JWKS nor JWKS URL.
18. Save the changes.
19. Select to import a certificate.
20. In the Archive format, select "JSON Web Key Set".
21. Select the file that was created earlier.
22. Click Import.  Another error occurred at this time.

### Anything else?

Keycloak is running on a container. The container was based on the latest tag of Keycloak.

The following is the version information shown at startup in the console:
![image](https://user-images.githubusercontent.com/47931389/171965976-cd5fe052-cc8b-48a1-999b-d278965f3e16.png)

**Comment by burner-account** [comment]
I have nothing to contribute towards the solution.
But i would be happy if you could improve step 2 of the guide to reproduce the error.
Not that some innocent soul gets the idea that submitting private keys to third party web services is the right thing to do.

**Comment by tnorimat** [comment]
@carlosli888 Hello,
Does this issue happen if you add ["use"](https://datatracker.ietf.org/doc/html/rfc7517#section-4.2) field like the following?
```
{
    "use": "sig",
    "kty": "EC",
    "kid": "client1-ES256",
    "crv": "P-256",
    "alg": "ES256",
    "x": "KhIuh2un6UWcBCIQqr5s3lSN42mrp5kjdf3JrasR1E4",
    "y": "bUIXyjZ6Q7-fLu-mp56OJjEHOAbGd3X30EMhS7SG-Vw"
}
```

**Comment by rmartinc** [comment]
@carlosli888 I think that @tnorimat is right. At least the `use` option is needed for sure. If you want you can configure it using a little project I have to do this: https://github.com/rmartinc/keycloak-jwks.
As I commented in the other issue the only way of configuring ES algorithm until version 19 is released is using the jwks options.

**Comment by carlosli888** [comment]
Thank you @tnorimat  and @rmartinc, adding "use" to the JWK is what was missed.  Based on [RFC7517, section-4.2](https://datatracker.ietf.org/doc/html/rfc7517#section-4.2), this attribute is optional unless required by the application.  Could we have the Keycloak documentation be more obvious as to what fields are required by Keycloak?  When I referenced the JWK standard, I added only the attributes as "required" by the specification.
